c++/src/capnp/compat/http-over-capnp.c++

```diff
@@ -504,6 +527,19 @@ public:
       return kj::NEVER_DONE;
     });
 
+    // We need to create a tlsStarter callback which sends a startTls request to the capnp server.
+    KJ_IF_MAYBE(tlsStarter, settings.tlsStarter) {
+      kj::Function<void(kj::StringPtr)> cb =
+          [remoteTlsStarter = pipeline.getTlsStarter()](kj::StringPtr expectedServerHostname) mutable -> void {
+        auto rpcRequest = remoteTlsStarter.startTlsRequest();
+        auto downPipe = kj::newOneWayPipe();
+        rpcRequest.setExpectedServerHostname(expectedServerHostname);
+
+        RemotePromise<capnp::TlsStarter::StartTlsResults> pipeline = rpcRequest.send();
```

Comment:
Er, this seems to be immediately dropping the promise returned by `send()`, which immediately cancels the call.

Unfortunately we might need to change the TLS starter callback to return `kj::Promise<void>` so that we rat return a promise here. And the caller will have to wait for that promise before doing any writes.